js/main.js

```diff
@@ -1,115 +1,34 @@
-                counter = counter-1;
-                removeCurrentAddActive(items);
-                removeActiveAddCurrent(items.previousSibling);
-            }else if(counter === 0){
-                items = document.getElementById('img'+counter);
-                removeCurrentAddActive(items);
-                counter = amount-1;
-                items = document.getElementById('img'+counter);
-                removeActiveAddCurrent(items);
-            }
-            navigate(-1);
-        });
-    }
-    function photoData(pics, myResp){
-        /**
-         * @param myResp          Information about the object.
-         * @param myResp.photos   Information about the object's members.
-         * @param myResp.photos.photo   Information about the object's members.
-         * @param myResp.photos.photo.farm   Information about the object's members.
-         * @param myResp.photos.photo.server   Information about the object's members.
-         * @param myResp.photos.photo.secret   Information about the object's members.
-         */
-        var farm, server, id, secret, picUrl;
-
-        for(var i=0; i< myResp.photos.photo.length; i++){
-            farm = myResp.photos.photo[i].farm;
-            server = myResp.photos.photo[i].server;
-            id = myResp.photos.photo[i].id;
-            secret = myResp.photos.photo[i].secret;
-            picUrl = "https://farm"+farm+".staticflickr.com/"+server+"/"+id+"_"+secret+"_z.jpg";
-            pics.push(picUrl);
-        }
-    }
-    function pupulate(pics){
-        for(var j=0; j<pics.length; j++){
-            var list = document.getElementById('list');
-            for(j = 0; j<pics.length; j++){
-                list.innerHTML = list.innerHTML + "<li class= 'active' id='img"+[j]+"'><img src='" + pics[j] + "' /></li>";
+    $(document).ready(function(){
+        (function loadPics(){
+            var api_key = '74e47a159e15cbcb6139ba9c9df64c13';
+            var method = "flickr.photos.search";
+            var user_id = "138698049@N03";
+            var format = '&format=json';
+            var url = "https://api.flickr.com/services/rest"+"?method="+method+"&api_key="+api_key+"&user_id="+user_id+format+'&nojsoncallback=1';
+
+            $.getJSON(url, function(data){
+                if (data) {
+                    createDomElm(data);
+                }
+            });
+            function createDomElm(data){
+                var farm, server, id, title, secret, picUrl, img, currentPhoto;
+
+                for(var i=0; i< data.photos.photo.length; i++){
+                    currentPhoto = data.photos.photo[i];
+
+                    farm = currentPhoto.farm;
+                    server = currentPhoto.server;
+                    id = currentPhoto.id;
+                    title = currentPhoto.title;
+                    secret = currentPhoto.secret;
+                    picUrl = "https://farm"+farm+".staticflickr.com/"+server+"/"+id+"_"+secret+"_z.jpg";
```

Comment:
Great! This looks much better. I like to do this anytime I have to reference by an index, and especially when there are so many child objects: `data.photos.photo[i].farm;`